lib/setup.py

```diff
@@ -47,10 +32,40 @@
     "All in pure Python. All for free."
 )
 
-pipfile = Project(chdir=False).parsed_pipfile
-
-packages = pipfile["packages"].copy()
-requirements = convert_deps_to_pip(packages, r=False)
+# IMPORTANT: We should try very hard *not* to add dependencies to Streamlit.
+# And if you do add one, make the required version as general as possible.
+# But include relevant lower bounds for any features we use from our dependencies.
+install_requires = [
```

Comment:
nit: maybe use uppercase (`INSTALL_REQUIRES`) here, since the other settings above also use uppercase notation.

Reply:
👍